debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/AbstractReplicationMessageColumn.java

```diff
@@ -0,0 +1,216 @@
+ * The metadata are parsed lazily.
+ * 
+ * @author Jiri Pechanec
+ *
+ */
+public abstract class AbstractReplicationMessageColumn implements ReplicationMessage.Column {
+
+    public static class TypeMetadata {
+        private static final Logger LOGGER = LoggerFactory.getLogger(TypeMetadata.class);
+
+        private static final int[] EMPTY_TYPE_MODIFIERS = {};
+        private static final Pattern TYPE_PATTERN = Pattern.compile("^(?<full>(?<base>[^(\\[]+)(?:\\((?<mod>.+)\\))?(?<suffix>.*?))(?<array>\\[\\])?$");
+        private static final Pattern TYPEMOD_PATTERN = Pattern.compile("\\s*,\\s*");
+            // "text"; "character varying(255)"; "numeric(12,3)"; "geometry(MultiPolygon,4326)"; "timestamp (12) with time zone"; "int[]"
+
+        /**
+         * The basic name of the type without constraints
+         */
+        private final String baseType;
+
+        /**
+         * The full name of the type including the constraints
+         */
+        private final String fullType;
+
+        /**
+         * The type constraints converted to their integer value
+         */
+        private final int[] typeModifiers;
+
+        /**
+         * True if the type is array
+         */
+        private final boolean isArray;
+
+        /**
+         * The shortened name of the type how would be reported by JDBC
+         */
+        private final String normalizedTypeName;
+
+        /**
+         * True if the type has not <code>NOT NULL</code> constraint
+         */
+        private final boolean optional;
+
+        public TypeMetadata(String columnName, String typeWithModifiers, boolean optional) {
+            this.optional = optional;
+            Matcher m = TYPE_PATTERN.matcher(typeWithModifiers);
+            if (!m.matches()) {
+                LOGGER.error("Failed to parse columnType for {} '{}'", columnName, typeWithModifiers);
+                throw new ConnectException(String.format("Failed to parse columnType '%s' for column %s", typeWithModifiers, columnName));
+            }
+            String fullType = m.group("full");
+            String baseType = m.group("base").trim();
+            if (!Objects.toString(m.group("suffix"), "").isEmpty()) {
+                baseType = String.join(" ", baseType, m.group("suffix").trim());
+            }
+            int[] typeModifiers = EMPTY_TYPE_MODIFIERS;
+            if (m.group("mod") != null) {
+                final String[] typeModifiersStr = TYPEMOD_PATTERN.split(m.group("mod"));
+                typeModifiers = new int[typeModifiersStr.length];
+                for (int i = 0; i < typeModifiersStr.length; i++) {
+                    try {
+                    typeModifiers[i] = Integer.parseInt(typeModifiersStr[i]);
+                    } catch (NumberFormatException e) {
+                        LOGGER.error("Failed to parse type modifier for {} '{}'", columnName, typeModifiersStr[i]);
```

Comment:
We should either log _or_ rethrow, but not both at the same time. I'll remove the logger call.